third_party/istio-1.0.2/istio.yaml

```diff
@@ -347,11 +347,11 @@ data:
       
       containers:
       - name: istio-proxy
-        # PATCH #2: Graceful shutdown of the istio-proxy.
+        # PATCH #2: Graceful shutdown of the istio-proxy. See https://github.com/istio/istio/issues/7136.
         lifecycle:
           preStop:
             exec:
-              command: ["sh", "-c", 'sleep 20; until curl -s localhost:15000/clusters | grep "inbound|80|" | grep "rq_active" | grep "rq_active::0"; do sleep 1; done;']
+              command: ["sh", "-c", 'sleep 20; while [ $(netstat -plunt | grep tcp | grep -v envoy | wc -l | xargs) -ne 0 ]; do sleep 1; done']
```

Comment:
@markusthoemmes So, when using our Maistra operator instead of Istio shipped w/ Knative, how would we go about getting these two Istio fixes in? These are something we need to wait for upstream Istio to pull in?

We can't just assume Knative owns the Istio it runs on top of.

Reply:
@bbrowning see https://jira.coreos.com/projects/SRVKS/issues/SRVKS-17 for more context on Maistra vs. Istio + fixes being done in upstream Istio.